build.gradle.kts

```diff
@@ -56,15 +73,16 @@ tasks {
     }
 }
 
-val repositoryUrl = if (version.toString().endsWith("SNAPSHOT"))
+val isSnapshot = version.toString().endsWith("-SNAPSHOT")
+
+val repositoryUrl = if (isSnapshot)
     "https://oss.sonatype.org/content/repositories/snapshots/"
 else
     "https://oss.sonatype.org/service/local/staging/deploy/maven2/"
 
 publishing {
     publications {
-        create<MavenPublication>("mavenJava") {
-            from(components["java"])
+        withType<MavenPublication> {
             groupId = project.group.toString()
             artifactId = project.name.toLowerCase()
             version = project.version.toString()
```

Comment:
The Kotlin multiplatform plugin will set these for you as there can be `Keval-jmv`, `Keval-js`, etc... If you remove these, the warning should go away

Reply:
Great, gonna check that later !